src/main/scala/amailp/intellij/robot/parser/SettingParser.scala

```diff
@@ -33,8 +33,7 @@ object SettingParser extends SubParser {
 
     def parseLibraryValueAndParameters() {
       consumeSeparator()
-      parseCellOfType(ast.LibraryValue)
-      parseRemainingCells()
+      parseRowOfType(ast.LibraryValue)
```

Comment:
@AmailP I doubt that this is done correctly and clear, so I need advice

Reply:
I do not get why the current implementation here is not enough.

As far as I can see currently we parse all the cells separately, and the proposal is to parse the whole row as a single `IElementType`. Am I correct?

Reply:
As a general concept, the plugin tries to separate all the various part of the `.robot` files during the parse phase (tables, rows, cells) and then infer later their meaning of the AST (Abstract Syntax Tree) and PSI tree.

If you have not already done it, I would suggest you to thoroughly study the following page:
https://www.jetbrains.org/intellij/sdk/docs/reference_guide/custom_language_support/implementing_parser_and_psi.html

It could probably be a good idea to have dedicated `PsiElement`s for the `WITH NAME` cell and for its parameter (the actual new name).

Reply:
And @skhomuti, woud you mind joining this gitter:
https://gitter.im/AmailP/robot-plugin

So, if you are interested, we can talk about more general topics like creating a release for your contributions.

Reply:
Exactly what is needed. I was thinking towards the new PsiElement, but I could not figure out the rules for parsing. The article should help.

A case that has not been handled in my current implementation is that there may be a comment at the end of the line. I will add a test for this.